### PR TITLE
ssl: Fix unused variable warnings

### PR DESCRIPTION
**Pull request by dlech**

### CODE REVIEW

**Comment by dlech** [comment]
I have packaged micropython as a debian package. The debian packaging tools automatically set the gcc option to treat warnings as errors. Since this library is not distributed with debian, I have to build it as part of the micropython package. The package will fail to build because of these errors.

**Comment by pfalcon** [comment]
This project is a set of (rebased) patches on top of the upstream axtls. The patches are optimized for minimal size. Moving code around aren't really supported (may/will lead to conflict on rebases). Moving previously patched-in lines to avoid warnings would be the most supported change, but such would need to be squashed into the original patch which introduced the change (=> a PR should contain enough info to perform such a squash).

Bottom line: applying this patch as is would be counter-productive to the maintenance of the fork. I might look in trying to assimilate these changes into the existing patchset - when time permits. Or feel to try the process as described above.

**Comment by pfalcon** [comment]
(And for your original usecase (a debian package) you always can use a standard solution for such a case - a distro package patch.)

**Comment by dlech** [comment]
I can certainly maintain my own patches. I am just trying to be a good OSS citizen and send my changes upstream. :smiley:
